strider/trapi.py

```diff
@@ -247,10 +247,58 @@ def filter_message(
                                         ] = message.knowledge_graph.edges[edge_id]
 
     message.results = kept_results
-    message.knowledge_graph.edges = kept_knowledge_graph.edges
+    if message.knowledge_graph:
+        message.knowledge_graph.edges = kept_knowledge_graph.edges
+    else:
+        message.knowledge_graph = kept_knowledge_graph
     message.auxiliary_graphs = kept_aux_graphs
 
 
+def clean_query_id(
+    message: Message,
+    curie_map: dict,
+    kp_id: str,
+    logger: logging.Logger = logging.getLogger(),
+):
+    """Remove any errant query_ids from result node_bindings."""
+    for result in message.results or []:
+        for qnode_id, node_bindings in result.node_bindings.items():
+            for node_binding in node_bindings:
+                if (
+                    node_binding.query_id is not None
+                    and node_binding.query_id
+                    not in message.query_graph.nodes[qnode_id].ids
+                ):
+                    # if query id exists and isn't in qgraph
```

Comment:
Does this actually solve the spoke problem?   The message that we send to spoke has DOID:7014 in the query graph.   It sends back a binding with query_id DOID:7014.  So I don't think this block ever gets invoked.

Reply:
This happens in the postproc after we convert the query_graph back to use BMT preferred prefixes, i.e. MONDO, so the query graph id and the `query_id` won't match.

Reply:
Does the knowledge_graph also get converted back at the same time?

Reply:
It does, along with result node binding ids

Reply:
Would it make sense to also convert query_id?

Reply:
Good point. I've moved this block up to within `fix_result` and figured out I could simplify the logic a little bit.